service/history/workflow/update/update.go

```diff
@@ -126,15 +126,21 @@ func newAccepted(id string, onComplete func(), opts ...updateOpt) *Update {
 
 func newCompleted(
 	id string,
-	fetchOutcome func(ctx context.Context) (*updatepb.Outcome, error),
+	fetchResult outcomeOrErr,
 	opts ...updateOpt,
 ) *Update {
+	makeOutcomeFuture := func() future.Future[*updatepb.Outcome] {
+		if fetchResult.err != nil {
+			return future.NewReadyFuture[*updatepb.Outcome](nil, fetchResult.err)
+		}
+		return future.NewReadyFuture(fetchResult.outcome, nil)
+	}
 	upd := &Update{
 		id:              id,
 		state:           stateCompleted,
 		instrumentation: &noopInstrumentation,
 		accepted:        future.NewReadyFuture[*failurepb.Failure](nil, nil),
-		outcome:         lazyOutcome(fetchOutcome),
+		outcome:         makeOutcomeFuture(),
```

Comment:
What's the point of making this function if we immediately call it?

Reply:
Symmetry with the contstructors above and below. It's more clear that all the same fields are assigned. If Go had ternary statement's I'd use one here. Alas

Reply:
I think we should just accept a future in this function then, so that we don't need the fetchResult structure.